### PR TITLE
Fix Stream cloning

### LINKED ISSUE #1154: InterpreterRelation interface iterators can't be copied
Souffle Interface iterators, generated from InterpreterRelation, don't like being copied. We get use after free errors when the original goes out of scope. I'm not seeing where the cloning goes wrong.

An example test, adapted from src/tests/interpreter_relation_test.cpp
```cpp
    // Set up an InterpreterRelation and interface
    Relation::iterator it = relInt.begin();
    // *it works as expected
    // Make a new iterator, move it to the first iterator, then let the new iterator go out of scope
    {
        Relation::iterator it2(relInt.begin());
        it = std::move(it2); // works
    }
    // *it works as expected

    // Now make a new iterator, copy it to the first iterator, then let the new iterator go out of scope
    {
        Relation::iterator it2(relInt.begin());
        it = it2; // isn't really a copy
    }
    // *it tries to use memory we've already freed
```

### PR DESCRIPTION
**Pull request by mmcgr**
Fixes #1154 

I don't really like the reload method I used, but haven't thought of a better approach

### CODE REVIEW

File: src/InterpreterIndex.cpp [hunk 1]
```diff
@@ -241,10 +249,20 @@ class GenericIndex : public InterpreterIndex {
             return c;
         }
 
+        int reload(TupleRef* out, int max) override {
+            int c = 0;
+            max = std::max(max, Stream::BUFFER_SIZE);
```
Comment (XiaowenHu96) [inline src/InterpreterIndex.cpp]: Did you mean std::min here?

Comment (mmcgr) [reply src/InterpreterIndex.cpp]: I did, yes.

Comment (mmcgr) [reply src/InterpreterIndex.cpp]: It should only be called by the Stream that it's coupled with, which already knows the limit, but there's nothing to enforce that so I check the size. At least, I do now that it uses std::min.

**Comment by mmcgr** [comment]
The appveyor test was accidentally applied here. It's not fully set up yet so should be ignored.

**Pull request merged**